telegram/_bot.py

```diff
@@ -308,6 +308,54 @@ def __init__(
 
         self._freeze()
 
+    async def __aenter__(self: BT) -> BT:
+        try:
```

Comment:
Having just written https://github.com/python-telegram-bot/python-telegram-bot/pull/3826#issuecomment-1714435152 I see that these methods do not have a docstring - although they are implicitly documented in the class docstring. If we decide that they should get a docstring, we can do that in a separate PR, just wanted your opinion on it :)

Reply:
yeah I think they should get a separate docstring, makes it more consistent.

Reply:
https://github.com/python-telegram-bot/python-telegram-bot/issues/3886